src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -76,8 +77,23 @@ public PluginManager(Jenkins jenkins) {
      * Force update the plugin update center metadata.
      */
     public void checkForUpdates() {
-        visit("checkUpdates");
-        waitFor(by.xpath("//span[@id='completionMarker' and text()='Done']"));
+        visit("advanced");
+        // The check now button is a POST with a redirect to the same page.
+        // We use the button itself to detect when the page has changed, which happens after the refresh has been done
+        final WebElement button = find(by.button("Check now"));
+        clickButton("Check now");
+        waitFor().withTimeout(30, TimeUnit.SECONDS).until(new Callable<Boolean>() {
```

Comment:
Note there is an overload where checked exceptions are treated as false. Plus, it gets reported in case the expected event never happen.